presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConnectorsAPI.java

```diff
@@ -34,37 +35,63 @@
 
 @Path("/connectors")
 @Api(description = "the connectors API")
-@javax.annotation.Generated(
-        value = "io.swagger.codegen.languages.JavaJAXRSSpecServerCodegen",
-        date = "2017-06-23T09:53:13.549-04:00")
 @Singleton
 public final class ConnectorsAPI
 {
     private static final APIFileHandler apiFileHandler = new APIFileHandler(Paths.get("/etc/presto/catalog"));
 
     @GET
-    @Produces({MediaType.TEXT_PLAIN})
+    @Produces(MediaType.TEXT_PLAIN)
     @ApiOperation(value = "Get avaiable connector file names")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "Retrieved configuration", response = String.class)})
+            @ApiResponse(code = 200, message = "Retrieved configuration")})
     public synchronized Response getConnectors()
     {
         return apiFileHandler.getFileNameList();
     }
 
     @GET
     @Path("/{file}")
-    @Produces({MediaType.TEXT_PLAIN})
-    @ApiOperation(value = "Get connectors by file")
+    @Produces(MediaType.TEXT_PLAIN)
+    @ApiOperation(value = "Get connector property by file")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "Retrieved file", response = String.class),
+            @ApiResponse(code = 200, message = "Retrieved file"),
             @ApiResponse(code = 404, message = "Resource not found")})
     public synchronized Response getConnectorFile(
             @PathParam("file") @ApiParam("The name of a file") String file)
     {
         return apiFileHandler.getFile(file);
     }
 
+    @GET
+    @Path("/{file}/{property}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @ApiOperation(value = "Get specific connector property")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Retrieved property"),
+            @ApiResponse(code = 404, message = "Resource not found")})
+    public synchronized Response getConnectorProperty(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            @PathParam("property") @ApiParam("A specific property") String property)
+    {
+        return apiFileHandler.getFileProperty(file, property);
+    }
+
+    @PUT
+    @Path("/{file}/{property}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @ApiOperation(value = "Update or create a certain property of connector file")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Property updated or created"),
+            @ApiResponse(code = 404, message = "Resource not found")})
+    public synchronized Response updateConnectorProperty(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            @PathParam("property") @ApiParam("A specific property") String property,
+            String value)
+    {
+        return apiFileHandler.updateProperty(file, property, value);
+    }
+
```

Comment:
Why'd you move `GET /file/property`? I think it would be better to put both of these functions after the operations on `/file`.